src/Extensions/AzureBlobStorage/Program.cs

```diff
@@ -1,155 +1,73 @@
-                result.Succeeded = true;
-            }
-            catch (Exception ex)
-            {
-                result.Succeeded = false;
-                result.FailureMessage = ex.Message;
-            }
-
-            string jsonBlob = JsonSerializer.Serialize<EgressArtifactResult>(result);
-            Console.Write(jsonBlob);
-
-            // return non-zero exit code when failed
-            return result.Succeeded ? 0 : 1;
-        }
-
-        private static AzureBlobEgressProviderOptions BuildOptions(ExtensionEgressPayload configPayload)
-        {
-            AzureBlobEgressProviderOptions options = GetOptions<AzureBlobEgressProviderOptions>(configPayload);
+            AzureBlobEgressProvider provider = new(logger);
 
-            // If account key was not provided but the name was provided,
-            // lookup the account key property value from EgressOptions.Properties
-            if (string.IsNullOrEmpty(options.AccountKey) && !string.IsNullOrEmpty(options.AccountKeyName))
+            Action<ExtensionEgressPayload, AzureBlobEgressProviderOptions> configureOptions = (configPayload, options) =>
             {
-                if (configPayload.Properties.TryGetValue(options.AccountKeyName, out string accountKey))
+                // If account key was not provided but the name was provided,
+                // lookup the account key property value from EgressOptions.Properties
+                if (string.IsNullOrEmpty(options.AccountKey) && !string.IsNullOrEmpty(options.AccountKeyName))
                 {
-                    options.AccountKey = accountKey;
+                    if (configPayload.Properties.TryGetValue(options.AccountKeyName, out string accountKey))
+                    {
+                        options.AccountKey = accountKey;
+                    }
+                    else
+                    {
+                        logger.EgressProviderUnableToFindPropertyKey(Constants.AzureBlobStorageProviderName, options.AccountKeyName);
+                    }
                 }
-                else
-                {
-                    Logger.EgressProviderUnableToFindPropertyKey(Constants.AzureBlobStorageProviderName, options.AccountKeyName);
-                }
-            }
 
-            // If shared access signature (SAS) was not provided but the name was provided,
-            // lookup the SAS property value from EgressOptions.Properties
-            if (string.IsNullOrEmpty(options.SharedAccessSignature) && !string.IsNullOrEmpty(options.SharedAccessSignatureName))
-            {
-                if (configPayload.Properties.TryGetValue(options.SharedAccessSignatureName, out string sasSig))
+                // If shared access signature (SAS) was not provided but the name was provided,
+                // lookup the SAS property value from EgressOptions.Properties
+                if (string.IsNullOrEmpty(options.SharedAccessSignature) && !string.IsNullOrEmpty(options.SharedAccessSignatureName))
```

Comment:
I'm assuming there is still validation happening on the dotnet-monitor side? Will the exceptions/logs thrown from the program be propagated to dotnet-monitor logs?

Reply:
stdout is logged [here](https://github.com/dotnet/dotnet-monitor/blob/feature/egressExtension/src/Tools/dotnet-monitor/Extensibility/ProgramExtension.OutputParser.cs#L83)
stderr is logged [here](https://github.com/dotnet/dotnet-monitor/blob/feature/egressExtension/src/Tools/dotnet-monitor/Extensibility/ProgramExtension.OutputParser.cs#L91)